support/src/main/java/com/shopify/graphql/support/AbstractResponse.java

```diff
@@ -100,7 +100,27 @@ protected JsonArray jsonAsArray(JsonElement element, String field) throws Schema
     }
 
     public List<Node> getNodes() {
-        return new ArrayList<>();
+        ArrayList<Node> children = new ArrayList<>();
+
+        if (this instanceof Node) {
+            children.add((Node) this);
+        }
+
+        for (String key : responseData.keySet()) {
+            collectNodes(get(key), children);
+        }
+
+        return children;
+    }
+
+    private void collectNodes(Object o, List<Node> collection) {
+        if (o instanceof AbstractResponse) {
+            collection.addAll(((AbstractResponse) o).getNodes());
```

Comment:
You could reduce the ArrayList allocations by moving the `if (this instanceof Node)` and `for (String key : responseData.keySet())` code in getNodes into this method so you can recursively call `collectNodes` with the existing collection instead.